assets/components/subscriptionsByCountryGroup/subscriptionsByCountryGroup.jsx

```diff
@@ -5,64 +5,334 @@
+
 import { type CountryGroupId } from 'helpers/internationalisation/countryGroup';
 import { type CommonState } from 'helpers/page/page';
+import { displayPrice } from 'helpers/subscriptions';
+import { addQueryParamsToURL } from 'helpers/url';
+import { getCampaign, type ReferrerAcquisitionData } from 'helpers/tracking/acquisitions';
+import {
+  getSubsLinks,
+  type SubsUrls,
+  iOSAppUrl,
+  androidAppUrl,
+  dailyEditionUrl,
+} from 'helpers/externalLinks';
+import { classNameWithModifiers } from 'helpers/utilities';
+import { countryGroups } from 'helpers/internationalisation/countryGroup';
+import { type ImageId } from 'helpers/theGrid';
+import { appStoreCtaClick } from 'helpers/tracking/googleTagManager';
 
 
 // ----- Types and State Mapping ----- //
 
 type PropTypes = {
   countryGroupId: CountryGroupId,
   headingSize: HeadingSize,
+  referrerAcquisitionData: ReferrerAcquisitionData,
 };
 
-
 function mapStateToProps(state: { common: CommonState }) {
 
   return {
     countryGroupId: state.common.internationalisation.countryGroupId,
+    referrerAcquisitionData: state.common.referrerAcquisitionData,
   };
 
 }
 
 
+// ----- Setup ----- //
+
+const appReferrer = 'utm_source=support.theguardian.com&utm_medium=subscribe_landing_page';
+const internationalAppReferrer = 'utm_source=support.theguardian.com&utm_medium=subscribe_landing_page&utm_campaign=international_subs_landing_pages';
+
+
 // ----- Component ----- //
 
 function SubscriptionsByCountryGroup(props: PropTypes) {
 
-  const { countryGroupId, headingSize, ...otherProps } = props;
+  const {
+    countryGroupId, headingSize, referrerAcquisitionData, ...otherProps
+  } = props;
+  const subsLinks = getSubsLinks(
+    countryGroupId,
+    referrerAcquisitionData.campaignCode,
+    getCampaign(referrerAcquisitionData),
+    referrerAcquisitionData,
+  );
+  const className = classNameWithModifiers(
+    'component-subscriptions-by-country-group',
+    [countryGroups[countryGroupId].supportInternationalisationId],
+  );
 
   if (countryGroupId === 'GBPCountries') {
     return (
-      <div {...otherProps}>
-        <DigitalSubscriptionsContainer
-          headingSize={headingSize}
-        />
-        <PaperSubscriptionsContainer
-          headingSize={headingSize}
-        />
+      <div className={className} {...otherProps}>
+        <Digital headingSize={headingSize} subsLinks={subsLinks} countryGroupId={countryGroupId} />
+        <Paper headingSize={headingSize} subsLinks={subsLinks} countryGroupId={countryGroupId} />
       </div>
     );
   }
 
   return (
-    <div {...otherProps}>
-      <InternationalSubscriptions
-        countryGroupId={countryGroupId}
-        headingSize={headingSize}
-        {...otherProps}
-      />
+    <div className={className} {...otherProps}>
+      <International headingSize={headingSize} subsLinks={subsLinks} />
     </div>
   );
 
 }
 
 
+// ----- Auxiliary Components ----- //
```

Comment:
I think these components are complex & numerous enough to break out into one or more separate files. This file is > 300 lines now which is quite big.